### ISSUE TITLE
[Feature] apply labels from nodegroup to the created asg to enable scale to zero for managed nodegroups

### ISSUE THREAD

**Issue by matti** [weaveworks/eksctl#4965]
### What feature/behavior/change do you want?

eksctl should apply all labels from created nodegroup to the ASG in k8s.io/cluster-autoscaler/node-template/label/NAME=VALUE format

### Why do you want this feature?

This enables scale to zero for managed nodegroups. See workaround here: https://github.com/aws/containers-roadmap/issues/724#issuecomment-1070687704

**Comment by TBBle** [comment]
Duplicate of #4853.

**Comment by Skarlso** [comment]
>Particularly for this request, editing the ASG underneath the managed node group introduces issues with updating the managed node group object, so I'm not sure eksctl should be automating that.

Yep. That's true.

**Comment by matti** [comment]
>editing the ASG underneath the managed node group introduces issues

what kind of issues? I'm running 3 clusters with these edits and seems to work great, should I be worried?

**Comment by Skarlso** [comment]
You're running fine the problem one when we were trying to do this through the managed nodegroup.

**Comment by matti** [comment]
?

**Comment by Skarlso** [comment]
Didn't you write that you are using some kind of workaround? Sorry, I might be confusing this issue with a different one.

**Comment by TBBle** [comment]
I couldn't actually find the discussion of the issue when I went looking, across the eksctl and aws containers roadmap issue trackers. "tag" and "label" are not highly-distinctive keywords.

The issue I'm recalling isn't a problem that affects the running cluster, it showed up with you [Edited a Managed NodeGroup's configuration](https://docs.aws.amazon.com/eks/latest/userguide/update-managed-node-group.html#mng-edit), as it would detect "ASG has been modified externally, refusing to apply the update". I said it was related to autoscaler tags, but thinking back, it might have been before we had LaunchTemplate support, and people were directly editing the ASG to change _that_ for a Managed NodeGroup.

It's possible this has been changed since then, and instead editing a MNG will apply only the delta (kind but of course carries risk of misbehaviour) or just blows away your own changes (unfortunate but not surprising given the API).
